ir/expression.def

```diff
@@ -404,21 +404,54 @@ class ConstructorCallExpression : Expression {
 
 /// Represents a list of expressions separated by commas
 class ListExpression : Expression {
+    bool defaultInitializer = false;
+    ListExpression(Util::SourceInfo srcInfo, IR::Vector<IR::Expression>& components, bool defaultInitializer)
+                   : Expression(srcInfo), defaultInitializer(defaultInitializer), components(components) {
+        validate();
+        initUnknown();
+    }
+    ListExpression(const IR::Type* type, const IR::Vector<IR::Expression>& components, bool defaultInitializer)
+                        : Expression(type), defaultInitializer(defaultInitializer), components(components) {
+        validate();
+        initUnknown();
+    }
+
     inline Vector<Expression> components;
     ListExpression {
         validate();
+        initUnknown();
+    }
+
+    void initUnknown(){
```

Comment:
In general the `type` field of an expression is not maintained by the compiler. It is only populated by TypeInference when invoked with a boolean flag.
Is this field used for the ListExpression?